lib/rails_translation_manager/locale_checker.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+module RailsTranslationManager
+  class LocaleChecker
+    attr_reader :locale_path
+
+    CHECKER_CLASSES = [MissingEnglishLocales,
+                       MissingForeignLocales,
+                       IncompatiblePlurals].freeze
+
+    def initialize(locale_path)
+      @locale_path = locale_path
+    end
+
+    def validate_locales
+      if locale_file_paths.compact.blank?
+        puts 'No locale files found for the supplied locale path'
+        return false
+      end
+
+      output_result
+    end
+
+    private
+
+    def output_result
+      if checker_errors.blank?
+        puts 'Locale files are in sync, nice job!'
+        true
+      else
+        checker_errors.each do |error_message|
+          puts "\n"
+          puts error_message
+          puts "\n"
+        end
+        false
+      end
+    end
+
+    def checker_errors
+      errors = CHECKER_CLASSES.flat_map do |checker|
```

Comment:
It should've been yep, good spot! Instead of memoizing, I've just used a variable instead: https://github.com/alphagov/rails_translation_manager/pull/12/files#diff-ce83bc1c5f9a3a9c6f81b6b6f308146f2f0d914cefa49fd78c87e10fd8324b87R26.